hamilton/graph_types.py

```diff
@@ -203,5 +203,6 @@ def version(self) -> str:
         Node hashes are in a sorted list, then concatenated as a string before hashing.
         To find differences between dataflows, you need to inspect the node level.
         """
-        sorted_node_versions = sorted([n.version for n in self.nodes])
+        sorted_nodes = sorted(self.nodes, key=lambda n: n.name)
```

Comment:
Let's add a unit test -- you can just create a node with a null version or ensure it has a function that doesn't give one.